app/models/spree/payment_method/stripe_credit_card.rb

```diff
@@ -81,6 +81,22 @@ def void(response_code, _creditcard, _transaction_options)
         gateway.void(response_code, {})
       end
 
+      def payment_intents_refund_reason
+        Spree::RefundReason.where(name: Spree::Payment::Cancellation::DEFAULT_REASON).first_or_create
+      end
+
+      def try_void(payment)
+        if v3_intents? && payment.completed?
+          Spree::Refund.create!(
+            payment: payment,
+            amount: payment.amount,
```

Comment:
I think you can use [credit_allowed](https://github.com/solidusio/solidus/blob/475d9db5d0291dd4aeddc58ec919988c336729bb/core/app/models/spree/payment.rb#L143-L145) here (like in [Spree::Payment::Cancellation#cancel](https://github.com/solidusio/solidus/blob/b8296bdb5c9651e60239ab27c0093636981fd0aa/core/app/models/spree/payment/cancellation.rb#L34))
I don't think it's a problem for Stripe but, if there are some partial refunds already done, `credit_allowed` should give the correct amount to be refunded. WDYT?